keel/api/v1/auth/views.py

```diff
@@ -97,22 +90,125 @@ def login(self, request, format="json"):
-            logger.error('ERROR: AUTHENTICATION:UserViewset ' + str(e))
+            logger.error('ERROR: AUTHENTICATION:LoginViewset ' + str(e))
             response['message'] = str(e)
             response['status'] = 0
             return Response(response, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
 
         data = {
             "email" : obj.user.email,
-            "token" : obj.token['token'],
+            "token" : obj.token,
         }
         response["message"] = data
         return Response(response, status=status.HTTP_200_OK)
 
+class GeneratePasswordReset(GenericViewSet):
+    serializer_class = serializers.GenerateTokenSerializer
+    
+    def token(self, request):
+        response = {
+            "status" : 1,
+            "message" : ""
+        }
+        serializer = self.serializer_class(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        email = serializer.validated_data['email']
+        current_time = str(datetime.datetime.now().timestamp()).split(".")[0]
+        
+        try:
+            user = User.objects.get(email=email)
+            obj, created = PasswordResetToken.objects.get_or_create(user=user, reset_token=current_time)
+            context = {
+                'token' : current_time
+            }
+            subject = 'Password Reset'
+            html_content = get_template('password_reset_email.html').render(context)
+            # send email
+            emails = EmailNotification(subject, html_content, [email])
+            emails.send_email()
+        except Exception as e:
+            logger.error('ERROR: AUTHENTICATION:GeneratePasswordReset ' + str(e))
+            response['message'] = str(e)
+            response['status'] = 0
+            return Response(response, status=status.HTTP_400_BAD_REQUEST)
+        
+        response["message"] = "Password Reset Link sent successfully"
+        return Response(response)
+
+
+class ConfirmPasswordReset(GenericViewSet):
+    serializer_class = serializers.ConfirmResetTokenSerializer
+
+    def confirm_reset(self, request):
+        response = {
+            "status" : 1,
+            "message" : ""
+        }
+        serializer = self.serializer_class(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        validated_data = serializer.validated_data
+        reset_token = validated_data.get('reset_token', None)
+        password = validated_data['password']
+        try:
+            token = PasswordResetToken.objects.get(reset_token=reset_token)
+            # get user and set password
+            user = User.objects.get(email=token.user)
+            user.set_password(password)
+            user.save()
+        except Exception as e:
+            logger.error('ERROR: AUTHENTICATION:ConfirmPasswordReset ' + str(e))
+            response['message'] = str(e)
+            response['status'] = 0
+            return Response(response, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
dont simploy throw a 400...
what if the db request failed
check for the type of exception and throw accordingly

Reply:
Should I create separate try, except?